hashring/hashring_test.go

```diff
@@ -23,15 +23,45 @@ package hashring
 import (
 	"fmt"
 	"math/rand"
+	"sort"
+	"sync"
 	"testing"
 
 	"github.com/dgryski/go-farm"
+	"github.com/uber/ringpop-go/events"
 
 	"github.com/stretchr/testify/assert"
 )
 
+// fake event listener
+type dummyListener struct {
```

Comment:
Could you use mockery here?  I think @dansimau has put a mock for `EventListener` in the events package. Don't know if that is in master yet.

Mocks made with mockery also have a function to count the number of calls that were made to a certain function of an interface.